main.js

```diff
@@ -116,4 +112,648 @@ let chahgeColor3 = f1();
+      city: 'Fortaleza',
+      country: 'Brazil',
+    },
+    {
+      name: 'Four Seasons Hotel',
+      city: 'Sydney',
+      country: 'Australia',
+    },
+    {
+      name: 'Le Meridien',
+      city: 'Nice',
+      country: 'France',
+    },
+    {
+      name: 'Apart Neptun',
+      city: 'Gdansk',
+      country: 'Poland',
+    },
+    {
+      name: 'Lux Isla',
+      city: 'Ibiza',
+      country: 'Spain',
+    },
+    {
+      name: 'Nox Hostel',
+      city: 'London',
+      country: 'UK',
+    },
+    {
+      name: 'Leonardo Vienna',
+      city: 'Vienna',
+      country: 'Austria',
+    },
+    {
+      name: 'Adagio Aparthotel',
+      city: 'Edinburgh',
+      country: 'UK',
+    },
+    {
+      name: 'Steigenberger Hotel',
+      city: 'Hamburg',
+      country: 'Germany',
+    },
+  ];
+
+function СountriesAndCities(x) { // создаем пустую функцию какую-то с каким-то аргументом х
+    const resultarr = {}; // результат константы не может быть изменен новым присваиванием
+const arrcountry = [...new Set(x.map((item) => item.country))]; // из аргумента х вытягиваем переменные стран и записывываем этот массив в константу arrcountry
+for  (let i=0; i<arrcountry.length; i++) { // для полученного массива стран arrcountry
+    const str7 = arrcountry[i]; // приравниваем массив стран к переменной str1
+    const newArr = x.filter((item) => item.country === str7); // из этого же аргумента х вытягиваем массив стран сравниваем его (===) с переменной str1 и совпадения записываем в новую константу newArr
+    resultarr[str7] = newArr.map((item) => item.city);// в константу resultarr записываем новый массив который будет к нашим совпадениям newArr вытягивать еще и города
+}
+return resultarr;
+}
+console.log(СountriesAndCities(Exercise7)) // вместо аргумента х записываем нужный нам массив
+
+
+
+
+////////lesson-7
+const obj1 = {
+    a: 'a',
+    b: {
+      a: 'a',
+      b: 'b',
+      c: {
+        a: 1,
+      },
+    },
+ };
+ const obj2 = {
+    b: {
+      c: {
+        a: 1,
+      },
+      b: 'b',
+      a: 'a',
+    },
+    a: 'a',
+ };
+ const obj3 = {
+    a: {
+      c: {
+        a: 'a',
+      },
+      b: 'b',
+      a: 'a',
+    },
+    b: 'b',
+ };
+
+ function deepEqual(obj1, obj2) { // Цикл через свойства объекта obj1
+    for (var p in obj1) { //Проверка на то, что оба объекта существуют
+      if (obj1.hasOwnProperty(p) !== obj2.hasOwnProperty(p)) return false;
+      switch (typeof (obj1[p])) {  // Глубокое сравнение объектов по ключам и значения:
+        case 'object':
+          if (!deepEqual(obj1[p], obj2[p])) return false;
+          break; 
+          // Сравнение данных типа function:
+        case 'function':
+          if (typeof (obj2[p]) == 'undefined' || (p != 'compare' && obj1[p].toString() != obj2[p].toString())) return false;
+          break;  // Сравнение значений:
+        default:
+          if (obj1[p] != obj2[p]) return false;
+      }
+    }
+    return true;
+  }
```

Comment:
try call deepEqual(obj2,obj3)

Reply:
Does not work. Do not understand why. Can you hear an explanation?